cmake/toolchain/llvm/Kconfig

```diff
@@ -17,3 +17,15 @@ config LLVM_USE_LLD
 	  Use LLVM built-in lld linker with llvm/clang.
 
 endchoice
+
+config LLVM_PROFILE
+	bool "Enable profiling output"
+	depends on ARCH_POSIX
+	help
+	  Builds the (posix) app with profiling and instruction
```

Comment:
A minor thing, The `Builds the (posix) app with..` can be a bit confusing.
It is everything we build (not just the app).
Maybe better to just say `Builds with profiling and instruction`
(You are mention below it only works for ARCH_POSIX)